app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,110 @@
+class VendorsController < ApplicationController
+  before_action :find_vendor, except: [:index]
+  before_action :require_login_match, except: [:index, :show]
+  before_action :find_orderitems, except: [:index, :show, :edit, :update]
+  before_action :tally_earnings, except: [:index, :show, :edit, :update]
+  before_action :tally_count, except: [:index, :show, :edit, :update]
+  before_action :render_orderitems, except: [:index, :show, :fulfillment, :edit, :update]
+
+  def index
+    @vendors = Vendor.all
+  end
+
+  def show;
+     render layout: 'spotlight'
+  end
+
+  def edit; end
+
+  def update
+    if login_match?
+      if @vendor.update(vendor_params)
+        flash[:success] = "Successfully updated profile."
+        redirect_to vendor_path(@vendor.id)
+      else
+        flash.now[:error] = "Error: Profile not updated."
+        render "edit"
+      end
+    end
+  end
+
+def fulfillment
+     render layout: 'fulfillment-template'
+end
+
+def fulfillment_pending
+     render 'fulfillment', layout: 'fulfillment-template'
+end
+
+def fulfillment_paid
+     render 'fulfillment', layout: 'fulfillment-template'
+end
+
+def fulfillment_shipped
+     render 'fulfillment', layout: 'fulfillment-template'
+end
+
+def fulfillment_cancelled
+     render 'fulfillment', layout: 'fulfillment-template'
```

Comment:
Seems like since each of these controller actions are the same, that you don't need separate controller actions for this action.